sihl/lib/core/core_jwt.mli

```diff
@@ -0,0 +1,21 @@
+type algorithm = Jwto.algorithm = HS256 | HS512 | Unknown
+
+type t
+
+type payload
+
+val empty : payload
+
+val add_claim : key:string -> value:string -> payload -> payload
+
+val encode : algorithm -> secret:string -> payload -> (string, string) result
+
+val decode : secret:string -> string -> (t, string) result
+
+val get_claim : key:string -> t -> string option
+
+val pp : Format.formatter -> t -> unit
+
+val eq : t -> t -> bool
+
```

Comment:
To me this signature looks quite odd. I would expect to build a data object that represents the token and then encode (probably using encoder) and then other direction to decode token (probably using decoder) into data structure. Then I would expect to see things groups into sensible modules that give me some hints about what each of them is used for.

Reply:
This would be the usage:

```
let token = 
Sihl.Jwt.empty
 |> Sihl.Jwt.add_claim ~key:"id" ~value:"bar"
 |> Sihl.encode Sihl.Jwt.HS256 ~secret:"foobar"
```

Decoding is also pretty straightforward.